### PR TITLE
testCommon junit5

### PR DESCRIPTION
**Pull request by barisoyoruk**
In [this issue](https://github.com/palantir/atlasdb/issues/6796), we are transitioning our test classes from JUnit4 to JUnit5.

In this pull request, we perform preliminary work to prepare the `timelock-server` package for JUnit5. For this reason, we are duplication `testCommon` package classes to **V2** which will support junit5. For more details, see the comments below.

### CODE REVIEW

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/TransactionPostMortemIntegrationTest.java [hunk 1]
```diff
@@ -1,122 +0,0 @@
...
-import java.util.function.Consumer;
-import org.junit.Before;
-import org.junit.Test;
-
-/**
- * TODO(fdesouza): Remove this once PDS-95791 is resolved.
- * @deprecated Remove this once PDS-95791 is resolved.
- */
-@Deprecated
-public class TransactionPostMortemIntegrationTest extends AbstractAsyncTimelockServiceIntegrationTest {
```
Comment (barisoyoruk) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/TransactionPostMortemIntegrationTest.java:54]: This ticket is resolved.

File: timelock-server/build.gradle [hunk 2]
```diff
@@ -169,20 +180,15 @@ dependencies {
...
-    stressTestImplementation 'com.palantir.tokens:auth-tokens'
-    stressTestImplementation 'com.github.tomakehurst:wiremock-standalone'
     stressTestImplementation project(':timelock-api:timelock-api-jersey')
+
     stressTestRuntimeOnly("org.junit.vintage:junit-vintage-engine") {
         because 'allows JUnit 3 and JUnit 4 tests to run'
     }
-
-    // TODO(fdesouza): Remove this once PDS-95791 is resolved.
-    integTestImplementation project(':examples:profile-client')
```
Comment (barisoyoruk) [inline timelock-server/build.gradle:185]: This ticket is resolved.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/DbKvsExtension.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import java.net.InetSocketAddress;
+import java.sql.Connection;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class DbKvsExtension implements BeforeAllCallback, AfterAllCallback {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/DbKvsExtension.java:40]: Exact replica of `DbKvsRule` but extension version for Junit5.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/NamespacedClientsV2.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+import com.palantir.lock.watch.LockWatchCache;
+import com.palantir.lock.watch.LockWatchCacheImpl;
+import com.palantir.timestamp.RemoteTimestampManagementAdapter;
+import com.palantir.timestamp.TimestampManagementRpcClient;
+import com.palantir.timestamp.TimestampManagementService;
+import com.palantir.timestamp.TimestampRange;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface NamespacedClientsV2 {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/NamespacedClientsV2.java:48]: Exact replica of `NamespacedClients` but uses other V2 implementations.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TemporaryConfigurationHolderV2.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import freemarker.template.Template;
+import freemarker.template.TemplateExceptionHandler;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Locale;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class TemporaryConfigurationHolderV2 implements BeforeAllCallback {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TemporaryConfigurationHolderV2.java:29]: Exact replica of `TemporaryConfigurationHolder` but extension version for Junit5.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockClusterV2.java [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class TestableTimelockClusterV2 implements BeforeAllCallback, AfterAllCallback {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockClusterV2.java:64]: Exact replica of `TestableTimelockCluster` but extension version for Junit5.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockServerV2.java [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+import com.palantir.timelock.config.PaxosInstallConfiguration.PaxosLeaderMode;
+import com.palantir.tokens.auth.AuthHeader;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class TestableTimelockServerV2 {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TestableTimelockServerV2.java:53]: Exact replica of `TestableTimelockServer` but uses V2 implementations.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TimeLockServerHolderV2.java [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+import io.dropwizard.jackson.Jackson;
+import io.dropwizard.testing.DropwizardTestSupport;
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public class TimeLockServerHolderV2 implements BeforeAllCallback, AfterAllCallback {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TimeLockServerHolderV2.java:45]: Exact replica of `TimeLockServerHolder` but extension version for Junit5.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TimeLockTestUtilsV2.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+import com.palantir.conjure.java.api.config.ssl.SslConfiguration;
+import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.immutables.value.Value;
+
+public final class TimeLockTestUtilsV2 {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/TimeLockTestUtilsV2.java:41]: Exact replica of `TimeLockTestUtils` but uses V2 implementations.

File: timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/util/TestProxiesV2.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+import com.palantir.conjure.java.config.ssl.SslSocketFactories;
+import com.palantir.conjure.java.config.ssl.TrustContext;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+public class TestProxiesV2 {
```
Comment (barisoyoruk) [inline timelock-server/src/testCommon/java/com/palantir/atlasdb/timelock/util/TestProxiesV2.java:36]: Exact replica of `TestProxies` but uses V2 implementations.

File: timelock-server/build.gradle (hunk 2 above)
Comment (LucasIME) [reply timelock-server/build.gradle:185]: I don't think the ticket is resolved? As in, it has been closed, but iirc, we never got to the bottom of the issue and solved it.

**Review by LucasIME — APPROVED** [review]

**Pull request merged**